### DISCUSSION TITLE
Store or return observations in Trace.tokenize()?

### DISCUSSION THREAD

**Discussion by e-cal** [AI-Planning/macq#70] [General]
Right now, `Trace.tokenize()` uses the given `Token` (`Observation` class) to get observation tokens for each step, and appends them to a list attribute. 

With `TraceList.tokenize()` we are going to return an `ObservationList` object, not store anything on an internal attribute.

So should `Trace.tokenize()` just return the list of observations instead of storing them? Or is there a good reason to store the list of observations on a `Trace`?

**Comment by haz** [comment]
Ya, I think it's safe to keep them separate for now. So return the `ObservationList`, and only let the acquisition techniques have access to that.
